### DISCUSSION TITLE
I'm mapping with the SLAMTEC LIDAR :)  Is there an urdf file for the Create3 with LIDAR and OAK-D Light ?

### DISCUSSION THREAD

**Discussion by jerrynsr** [iRobotEducation/create3_docs#66] [Questions] [answered]
Hi,
I was wondering if an  urdf file exists for the iRobot Create3.  The LIDAR example works great, now I would like to move on to NAV2.  Should I search for the Turtlebot 4 examples and start from there ?

**Comment by shamlian** [comment, marked as answer]
There is a urdf file for [Create 3](https://github.com/iRobotEducation/create3_sim/tree/637f02b9f7ddcc28de35e5e003c998290a51fccd/irobot_create_common/irobot_create_description/urdf), but we have not published one with arbitrary sensors. You don't need a urdf on a real robot; you can use the `tf` frames that Create 3 publishes in combination with extra publishers you write to reference your sensors' coordinate frames to the robot.

**Comment by jerrynsr** [comment]
Is there a good reference to use to convert a urdf defined for Gazebo
for use with a physical robot ?

**Reply by justinIRBT** [reply]
There is a `robot_state_publisher` described here:
https://docs.ros.org/en/foxy/Tutorials/URDF/Using-URDF-with-Robot-State-Publisher.html
but I think that might be overkill for what you are trying to accomplish.
The easiest path to get the camera into your TF tree would be to just use a `static_transform_publisher` similar to what was done for the laser here:
https://github.com/iRobotEducation/create3_examples/blob/galactic/create3_lidar/launch/sensors_launch.py#L11
I haven't used the OAK-D, but something to watch out for is which tf frames the OAK-D driver is publishing it's data with respect to (what is in the message header.frame_id), as the image frame might be rotated with respect to the robot.  As an example, when working with the Intel Realsense, I had to use the following frames:
```
Node(package='tf2_ros', executable='static_transform_publisher',
            arguments=['-0.03', '0.0', '0.19', '0', '0.03', '0.0', 'base_link', 'camera_link']),
Node(package='tf2_ros', executable='static_transform_publisher',
            arguments=['0.0', '0.03', '0.0', str(-math.pi/2), '0', str(-math.pi/2), 'camera_link', 'camera_depth_optical_frame']),
```
Where the `camera_depth_optical_frame` was published by the realsense driver and had a 90 degree offset.  Combining the `base_link`->`camera_link` modeling the camera's physical position on the robot with `camera_link`->`camera_depth_optical_frame` put the depth data at the correct position/orientation in `base_link`.

**Reply by alsora** [reply]
+1 on this.
You don't need urdf on real robots.

The create 3 is already publishing TF frames describing the robot, you only need to create extra publishers for the accessories/frame that you attached to it.

**Comment by jerrynsr** [comment]
I'm looking for how to add the Slamtec a-1 to the urdf
